defi/src/api2/db/index.ts

```diff
@@ -116,10 +129,21 @@ async function _getLatestProtocolItem(ddbPKFunction: Function, protocolId: strin
   return item.data
 }
 
-async function _getClosestProtocolItem(ddbPKFunction: Function, protocolId: string, timestamp: number) {
+async function _getClosestProtocolItem(ddbPKFunction: Function, protocolId: string, timestampTo: number, { searchWidth, timestampFrom }: { searchWidth?: number, timestampFrom?: number }) {
   const table = getTVLCacheTable(ddbPKFunction)
+  let timestampFilter: any = { [Op.lte]: timestampTo }
+
+  if (searchWidth && timestampFrom)
+    throw new Error('Cannot use both searchWidth and timestampFrom')
+
+  // this has a bug, it will return the latest item not closest to timestampTo
+  if (searchWidth)
```

Comment:
a simple solution here is to get these results from sql and then run some js code over them that picks the one with the closest timestamp, its what i was doing with ddb

Reply:
the issue is this pg query/call returns only one result. Will a different query specifically for this